File: certificates/problem-solving-basic/subarray-sums/findSum.js
function findSum(numbers, queries) {
  let a = [0];
  let b = [0];
  for (let x of numbers) {
    a.push(a[a.length - 1] + x);
    b.push(b[b.length - 1] + (x == 0));
  }
  return queries.map(([l, r, x]) => a[r] - a[l - 1] + x * (b[r] - b[l - 1])
  );
}

File: certificates/problem-solving-basic/subarray-sums/findSum.c
#include <stdio.h>
long* findSum(int numbers_count, int* numbers, int queries_rows, int queries_columns, int** queries, int* result_count)
{

  long *result = (long *)malloc(queries_rows * sizeof(long));
  long a[numbers_count + 1];
  long b[numbers_count + 1];
  a[0] = 0;
  b[0] = 0;

  for (int i = 0; i < numbers_count; i++)
  {
    a[i + 1] = a[i] + numbers[i];
    b[i + 1] = b[i] + (numbers[i] == 0);
  }

  for (int i = 0; i < queries_rows; i++)
  {
    int x = queries[i][2];
    int r = queries[i][1];
    int l = queries[i][0];

    result[i] = a[r] - a[l - 1] + x * (b[r] - b[l - 1]);
  }
  
  *result_count=queries_rows;
  return result;
}